raiden/network/transport/matrix/transport.py

```diff
@@ -959,28 +1042,44 @@ def _validate_matrix_messages(
                 raise RaidenUnrecoverableError(f"Received message in room {room.canonical_alias}.")
 
             # XXX-UAM: Whitelist check was here
+
+            sender_metadata: AddressMetadata = dict(user_id=UserID(sender_id))
             if is_raiden_message:
-                raiden_messages.extend(
-                    validate_and_parse_message(message["content"]["body"], peer_address)
-                )
+                for parsed_message in validate_and_parse_message(
+                    message["content"]["body"], peer_address
+                ):
+                    raiden_message = ReceivedRaidenMessage(
+                        message=parsed_message,
+                        sender=peer_address,
+                        sender_metadata=sender_metadata,
+                    )
+                    raiden_messages.append(raiden_message)
             if is_call_message:
-                call_messages.append((peer_address, message))
-
+                call_message = ReceivedCallMessage(
+                    message=message,
+                    sender=peer_address,
+                )
+                call_messages.append(call_message)
         return raiden_messages, call_messages
 
-    def _process_messages(self, all_messages: List[Message]) -> None:
+    def _process_messages(self, all_messages: List[ReceivedRaidenMessage]) -> None:
```

Comment:
While you're touching this anyway it may be a good idea to rename this to `_process_raiden_messages()` to be consistent with `_process_call_messages()`.